### PR TITLE
FOUR-10447: Add Preview for screens

### PR DESCRIPTION
**Pull request by danloa**
## Issue & Reproduction Steps
Adds the Preview for Screens. This preview will be used in modeler.

image

## Related Tickets & Packages
- [https://processmaker.atlassian.net/browse/FOUR-10447](https://processmaker.atlassian.net/browse/FOUR-10447)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-f8cc5d1624.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-f8cc5d1624.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-f8cc5d1624.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-f8cc5d1624.eng.processmaker.net

File: src/components/inspectors/PreviewPanel.vue [hunk 1]
```diff
@@ -107,7 +108,19 @@ export default {
...
 
+      const defaultDataTransform = (node) => Object.entries(node.definition).reduce((data, [key, value]) => {
+        data[key] = value;
+        return data;
+      }, {});
+
+      this.data = defaultDataTransform(this.highlightedNode);
+      // this.data = type && this.nodeRegistry[type].inspectorData
+      //   ? this.nodeRegistry[type].inspectorData(this.highlightedNode, defaultDataTransform, this)
+      //   : defaultDataTransform(this.highlightedNode);
```
Comment (josechirivella) [inline src/components/inspectors/PreviewPanel.vue:119]: remove

File: src/components/inspectors/PreviewPanel.vue [hunk 2]
```diff
@@ -57,11 +57,11 @@
     </b-row>
 
     <b-row>
-      <div class="item-title"> {{ screenTitle }} </div>
       <div class="task-title"> {{ taskTitle }} </div>
     </b-row>
 
-    <no-preview-available/>
+    <no-preview-available v-if="previewUrl === null"/>
+    <iframe v-if="previewUrl !== null" :src="previewUrl" style="width:100%; height:100%;border:0px none;"/>
```
Comment (josechirivella) [inline src/components/inspectors/PreviewPanel.vue:64]: v-else here or `!!previewUrl`

**Review by josechirivella — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-f8cc5d1624.eng.processmaker.net

File: src/components/inspectors/PreviewPanel.vue (hunk 1 above)
Comment (danloa) [reply src/components/inspectors/PreviewPanel.vue:119]: removed

File: src/components/inspectors/PreviewPanel.vue (hunk 2 above)
Comment (danloa) [reply src/components/inspectors/PreviewPanel.vue:64]: done

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-f8cc5d1624.eng.processmaker.net

**Review by josechirivella — APPROVED** [review]

File: src/components/inspectors/PreviewPanel.vue [hunk 3]
```diff
@@ -57,11 +57,26 @@
...
+          </g>
+        </g>
+      </svg>
```
Comment (josechirivella) [inline src/components/inspectors/PreviewPanel.vue:75]: can you move this svg into his own file?

File: src/components/inspectors/PreviewPanel.vue [hunk 4]
```diff
@@ -57,11 +57,26 @@
...
+    </div>
+
+    <no-preview-available v-show="!previewUrl"/>
+    <iframe title="Preview" v-show="!!previewUrl && !showSpinner" :src="previewUrl" style="width:100%; height:100%;border:0px none;" @load="loading"/>
```
Comment (josechirivella) [inline src/components/inspectors/PreviewPanel.vue:79]:
```suggestion
    <iframe title="Preview" v-show="!!previewUrl && !showSpinner" :src="previewUrl" style="width:100%; height:100%;border: none;" @load="loading"/>
```

File: src/components/inspectors/PreviewPanel.vue [hunk 5]
```diff
@@ -120,8 +154,25 @@ export default {
...
+    previewNode(force = false) {
+      if (!this.highlightedNode || (!this.visible && !force)) {
+        return;
+      }
+
+      const previewConfig = this.previewConfigs.find(config => {
+        return config.matcher(this.data);
+      });
+
+      var clone = {};
```
Comment (josechirivella) [inline src/components/inspectors/PreviewPanel.vue:166]: don't use var anymore, let if you are going to override the value of the variable later on
```suggestion
      let clone = {};
```

File: src/components/inspectors/PreviewPanel.vue [hunk 6]
```diff
@@ -120,8 +154,25 @@ export default {
...
+      for (let prop in this.data) {
+        if ((previewConfig?.receivingParams ?? []).includes(prop)) {
+          clone[prop] = this.data[prop];
+        }
+      }
+      const nodeData = encodeURI(JSON.stringify(clone));
+
+      this.previewUrl = previewConfig ? `${previewConfig.url}?node=${nodeData}` : null;
+      this.taskTitle = this?.highlightedNode?.definition?.name;
```
Comment (josechirivella) [inline src/components/inspectors/PreviewPanel.vue:175]: `this` in this case will always be defined. No need to add the first `?`
```suggestion
      this.taskTitle = this.highlightedNode?.definition?.name;
```

**Review by josechirivella — CHANGES_REQUESTED** [review]

**Review by josechirivella — APPROVED** [review]

**Pull request merged**